### PR TITLE
[crowdstrike] Extract user and host names from the name field

### PR DESCRIPTION
**Pull request by chrisberkhout**
## Proposed commit message

```
[crowdstrike] Extract user and host names from the name field (#)

Some alerts data from Crowdstrike has a `name` field of the form
`USERNAME on HOSTNAME`. Extract those values when present, add them to
`related.user` and `related.hosts`, and use them for `user.name` and
`host.name` if those fields aren't already populated.

Also corrects one use of `related.hash` to be `related.hosts`.
```

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices)

### CODE REVIEW

File: packages/crowdstrike/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,12 @@
 # newer versions go on top
+- version: "1.46.0"
+  changes:
+    - description: Extract user and host names from the name field.
```
Comment (kcreddy) [inline packages/crowdstrike/changelog.yml:4]:
```suggestion
    - description: Extract user and host names from the alert.name field.
```

**Review by kcreddy — APPROVED** [review]
nit. LGTM

File: packages/crowdstrike/changelog.yml (hunk 1 above)
Comment (chrisberkhout) [reply packages/crowdstrike/changelog.yml:4]: It's a top level field in the incoming data. Renamed from `json.name` before the new logic.

Comment (kcreddy) [reply packages/crowdstrike/changelog.yml:4]: Yeah, I deduced that (a bit late).

**Pull request merged**